flyteadmin/auth/authzserver/metadata_provider.go

```diff
@@ -72,9 +80,14 @@ func (s OAuth2MetadataProvider) GetOAuth2Metadata(ctx context.Context, r *servic
 			httpClient.Transport = transport
 		}
 
-		response, err := httpClient.Get(externalMetadataURL.String())
+		response, err := sendAndRetryHttpRequest(httpClient, externalMetadataURL.String(), s.cfg.AppAuth.ExternalAuthServer.RetryAttempts, s.cfg.AppAuth.ExternalAuthServer.RetryDelayMilliseconds.Duration)
 		if err != nil {
-			return nil, err
+			if response != nil {
+				logger.Errorf(ctx, "Failed to get oauth metadata. Error code: %v. Err: %v", response.StatusCode, err)
+				return nil, flyteErrors.NewFlyteAdminError(codes.Code(response.StatusCode), "Failed to get oauth metadata.")
+			}
```

Comment:
Are we expected to see a non nil response on err, I would consider not considering this case, we have special handling around responses in error cases

Reply:
Not expected! Just wanted to add an edge case here, happy to remove it though! I did not know we had special handling around responses in error cases :)

Reply:
In this particular case, you are getting the statusCode from the response so we can keep it. 
Also on L90 , 500 is not a valid grpc error code, I would use code.Internal there